storage/remote/queue_manager_test.go

```diff
@@ -129,20 +141,33 @@ func TestSpawnNotMoreThanMaxConcurrentSendsGoroutines(t *testing.T) {
 	m := NewStorageQueueManager(c, n)
 
 	go m.Run()
+	defer m.Stop()
 
 	for _, s := range samples {
 		m.Append(s)
 	}
 
-	for i := 0; i < maxConcurrentSends; i++ {
-		c.getData <- true // Wait while all goroutines are spawned.
+	// Wait until the Run() loop drains the queue.  If things went right, it
+	// should then immediately block in sendSamples(), but, in case of error,
+	// it would spawn too many goroutines, and thus we'd see more calls to
+	// client.Store()
+	//
+	// The timed wait is maybe non-ideal, but, in order to verify that we're
+	// not spawning too many concurrent goroutines, we have to wait on the
+	// Run() loop to consume a specific number of elements from the
+	// queue... and it doesn't signal that in any obvious way, except by
+	// draining the queue.  Also, this way, if something goes surprisingly
+	// wrong that prevents the queue from being drained (aka, not just spawning
+	// extra goroutines, but something totally unexpected), we deadlock, which
+	// is a clear signal of a problem.
+	for len(m.queue) > 0 {
+		time.Sleep(10 * time.Millisecond)
```

Comment:
Hum, I'd read more about the code first. But should we have a timeout for that, so that if things go completely wrong, we abort after something like 1s or whatever makes sense?